tsdb/tsm1/engine_measurement_notime_schema.go

```diff
@@ -259,38 +296,56 @@ func (e *Engine) MeasurementTagKeysNoTime(ctx context.Context, orgID, bucketID i
 }
 
 func (e *Engine) tagKeysNoTime(ctx context.Context, orgID, bucketID influxdb.ID, predicate influxql.Expr) (cursors.StringIterator, error) {
+	span, ctx := tracing.StartSpanFromContext(ctx)
+	defer span.Finish()
+
 	if err := ValidateTagPredicate(predicate); err != nil {
 		return nil, err
 	}
 
 	orgBucket := tsdb.EncodeName(orgID, bucketID)
 
-	vals := make([]string, 0, 32)
+	var (
+		vals              = make([]string, 0, 32)
```

Comment:
Is this determined empirically or is there some compile-time known reason for it to be 32? Perhaps a comment or moving to a constant would clarify.

Reply:
It was chosen over the default of 0 for an empty slice, which grows to 1, 2, 4, 8, 16 and eventually 32. The expectation is there will be typically more than 8 values, avoiding at least a number of allocations and copies.

https://play.golang.org/p/y-8AkpU-w-x

Reply:
I will add a comment to clarify 👍